lib/puppet/rest/routes.rb

```diff
@@ -34,11 +34,14 @@ def self.get_certificate(client, name)
     # HTTP client. Accepts a block to stream responses to disk.
     # @param [Puppet::Rest::Client] client the HTTP client to use to make the request
     # @param [String] name the crl to fetch
+    # @param [String] last_modified the httpdate of when the agent crl was last modified
     # @raise [Puppet::Rest::ResponseError] if the response status is not OK
     # @return nil
-    def self.get_crls(client, name, &block)
+    def self.get_crls(client, name, last_modified, &block)
       ca.with_base_url(client.dns_resolver) do |base_url|
-        header = { 'Accept' => 'text/plain', 'Accept-Encoding' => ACCEPT_ENCODING }
+        header = { 'Accept' => 'text/plain',
+                   'Accept-Encoding' => ACCEPT_ENCODING,
+                   'If-Modified-Since' => last_modified }
```

Comment:
Do you prefer I send the header with nil value (if there is not yet a crl), or  NOT send this "If-Modified-Since" header at all if its nil?  What's better practice?